spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/RedisHealthIndicator.java

```diff
@@ -45,12 +51,30 @@ protected void doHealthCheck(Health.Builder builder) throws Exception {
 				.getConnection(this.redisConnectionFactory);
 		try {
 			Properties info = connection.info();
-			builder.up().withDetail("version", info.getProperty("redis_version"));
+			if (connection instanceof RedisClusterConnection) {
+				redisClusterInfo(builder,
+						((RedisClusterConnection) connection).clusterGetNodes(), info);
+			}
+			else {
+				defaultRedisInfo(builder, VERSION, info.getProperty(REDIS_VERSION));
+			}
 		}
 		finally {
 			RedisConnectionUtils.releaseConnection(connection,
 					this.redisConnectionFactory);
 		}
 	}
 
+	private void defaultRedisInfo(Health.Builder builder, String key, String value) {
+		builder.up().withDetail(key, value);
+	}
+
+	private void redisClusterInfo(Health.Builder builder,
+			Iterable<RedisClusterNode> redisNodes, Properties info) {
+		for (RedisNode redisNode : redisNodes) {
+			defaultRedisInfo(builder, redisNode.asString() + "." + VERSION,
+					info.getProperty(redisNode.asString() + "." + REDIS_VERSION));
```

Comment:
I'm not too keen on the structure of the JSON, both because the keys vary and because it's not very extensible. I'd prefer something like this:

```json
{
    "status": "UP",
    "nodes": [
        {
            "address": "127.0.0.1:7001",
            "version": "3.0.7"
        },
        {
            "address": "127.0.0.1:7002",
            "version": "3.0.7"
        },
        {
            "address": "127.0.0.1:7003",
            "version": "3.0.7"
        }
    ]
}
```

The keys are the same for every node and we can also easily add extra information about a node.